src/Qir/Execution/Tools/Executable/QirFullStateExecutable.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Quantum.Qir.Tools.Driver;
+using Microsoft.Quantum.Qir.Utility;
+
+namespace Microsoft.Quantum.Qir.Tools.Executable
+{
+    /// <summary>
+    /// Class to create and run QIR-based executables that use the full-state simulator.
+    /// </summary>
+    public class QirFullStateExecutable : QirExecutable
+    {
+        public QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger, IQirDriverGenerator driverGenerator, IQirExecutableGenerator executableGenerator, IQuantumExecutableRunner runner)
+            : base(executableFile, qirBytecode, logger, driverGenerator, executableGenerator, runner)
+        {
+
+        }
+
+        public static IQirExecutable CreateInstance(FileInfo executableFile, byte[] qirBytecode, ILogger logger)
```

Comment:
I think a `static` method is not needed in this case. I would prefer to have a constructor that accepts an `ILogger` argument:
```csharp
public QirFullStateExecutable(FileInfo executableFile, byte[] qirBytecode, ILogger logger)
{
}
```